### ISSUE TITLE
Failed to package for Android with OpenCV. (scikit-build could not get a working generator for your system)

### ISSUE THREAD

**Issue by Pairman** [beeware/briefcase#1587]
### Describe the bug

I'm using Ubuntu x86_64 to package my app for Android which uses OpenCV. But it keeps failing.

I have had ```build-essentials``` installed. But I guess CMake is probably asking for a cross-compiler. Is there any way to use dockcross for android packaging, or is there any other fixes? Thanks in advance.

Error message:
```
             -- Trying 'Unix Makefiles' generator - failure                                                                                                        subprocess.py:685
             --------------------------------------------------------------------------------                                                                      subprocess.py:685
                                                                                                                                                                   subprocess.py:685
                             ********************************************************************************                                                      subprocess.py:685
                             scikit-build could not get a working generator for your system. Aborting build.                                                       subprocess.py:685
                                                                                                                                                                   subprocess.py:685
                             Building Linux wheels for Python 3.10 requires a compiler (e.g gcc).                                                                  subprocess.py:685
             But scikit-build does *NOT* know how to install it on ubuntu                                                                                          subprocess.py:685
                                                                                                                                                                   subprocess.py:685
             To build compliant wheels, consider using the manylinux system described in PEP-513.                                                                  subprocess.py:685
             Get it with "dockcross/manylinux-x64" docker image:                                                                                                   subprocess.py:685
                                                                                                                                                                   subprocess.py:685
               https://github.com/dockcross/dockcross#readme                                                                                                       subprocess.py:685
                                                                                                                                                                   subprocess.py:685
             For more details, please refer to scikit-build documentation:                                                                                         subprocess.py:685
                                                                                                                                                                   subprocess.py:685
               http://scikit-build.readthedocs.io/en/latest/generators.html#linux                                                                                  subprocess.py:685
                                                                                                                                                                   subprocess.py:685
                             ********************************************************************************                                                      subprocess.py:685
             ----------------------------------------                                                                                                              subprocess.py:685
           ERROR: Failed to install opencv-python from                                                                                                             subprocess.py:685
           https://files.pythonhosted.org/packages/25/72/da7c69a3542071bf1e8f65336721b8b2659194425438d988f79bc14ed9cc/opencv-python-4.9.0.80.tar.gz#sha256=1a9f0e6                  
           267de3a1a1db0c54213d022c7c8b5b9ca4b580e80bdc58516c922c9e1 (from -r requirements.txt (line 4)).                                                                           
           For assistance, please raise an issue at https://github.com/chaquo/chaquopy/issues.                                                                     subprocess.py:685
           Chaquopy: Exit status 1                                                                                                                                 subprocess.py:685
                                                                                                                                                                   subprocess.py:685
           > Task :app:generateDebugPythonRequirements FAILED                                                                                                      subprocess.py:685
                                                                                                                                                                   subprocess.py:685
           FAILURE: Build failed with an exception.                                                                                                                subprocess.py:685
```

### Steps to reproduce

1. Add dependency ```opencv-python``` for project.
2. Run ```briefcase package android -p apk``` on Ubuntu x86_64.

### Expected behavior

Build successful.

### Environment

OS: Ubuntu 22.04 x86_64
Python: 3.10.12
pip: 23.3.1
briefcase: 0.3.16
toga: 0.4.0

### Logs

[briefcase.2024_01_01-13_25_30.package.log](https://github.com/beeware/briefcase/files/13803735/briefcase.2024_01_01-13_25_30.package.log)

**Comment by Pairman** [comment]
Update: This was solved by downgrading Python to 3.8.

**Comment by vbnmjj** [comment]
thanks
